realm-jni/src/util.cpp

```diff
@@ -131,6 +131,11 @@ void ThrowException(JNIEnv* env, ExceptionKind exception, const std::string& cla
             jExceptionClass = env->FindClass("java/lang/IllegalStateException");
             message = "Illegal State: " + classStr;
             break;
+
+        case UnreachableVersion:
+            jExceptionClass = env->FindClass("io/realm/internal/async/UnreachableVersionException");
+            message = "Handover failed due to version mismatch: " + classStr;
```

Comment:
The message propagated to the user in the onError callback would end up to be "Handover failed due to version mismatch: begin_read failed" as far as I could tell? Would it be possible to make that more user friendly? e.g. is there a way to prevent it from happening?